lib/claim_letters/claim_letter_downloader.rb

```diff
@@ -41,10 +41,23 @@ def get_letter(document_id)
     end
 
     def format_letter_data(docs)
-      letters = docs.select { |d| d[:doc_type] == '184' }
-      letters.sort_by do |d|
-        DateTime.strptime(d[:upload_date], '%a, %d %b %Y')
-      end.reverse
+      letters = Marshal.load(Marshal.dump(docs))
+      letters = letters.select { |d| d[:doc_type] == '184' }
+      letters = letters.sort_by(&:upload_date).reverse
+      letters = letters.map do |d|
+        d[:upload_date] = DateTime.parse(d[:upload_date].to_s).strftime('%a, %d %b %Y')
+        d[:received_at] = DateTime.parse(d[:received_at].to_s).strftime('%a, %d %b %Y')
+        d
+      end
+      letters.map(&:marshal_dump)
+    end
+
+    private
+
+    def file_number
+      return @user.ssn unless @user.participant_id.present?
```

Comment:
Can you add a comment here indicating that this shouldn't happen in prod? Anyone else who encounters this may not understand why we are accounting for an ID that should be there in prod

Reply:
Good point, will add.